src/common/OperationsBase.ts

```diff
@@ -55,22 +55,61 @@ export class OperationsBase {
     let response: Response | undefined;
 
     for (let attempt = 1; attempt <= MAX_ATTEMPTS; attempt++) {
-      response = await this.fetch(
-        nextUrl,
-        requestOptions,
-      );
+      try {
+        response = await this.fetch(
+          nextUrl,
+          requestOptions,
+        );
+      } catch (error) {
+        if ("code" in (error as any)) {
+          switch ((error as any).code) {
+            case "ETIMEDOUT":
+            case "ECONNRESET":
+            case "EADDRINUSE":
+            case "ECONNREFUSED":
+            case "EPIPE":
+            case "ENOTFOUND":
+            case "ENETUNREACH":
+            case "EAI_AGAIN":
+            case "ECONNABORTED":
+              await new Promise((resolve) => setTimeout(resolve, (2 ** attempt) * 1000));
+              continue;
+            default: throw error;
+          }
+        }
+
+        throw error;
+      }
+
+      if (undefined === response)
+        throw new Error("Unknown error has occurred while executed the request.");
 
       if (response.status >= 200 && response.status < 300) {
         return response;
-      } else if (attempt < MAX_ATTEMPTS && 429 === response.status) {
-        const retryAfter = response.headers.get("Retry-After");
-        if (null === retryAfter) {
-          throw response;
+      } else if (attempt < MAX_ATTEMPTS) {
+        switch (response.status) {
+          case 429:
+            const retryAfter = response.headers.get("Retry-After");
+            if (null === retryAfter) {
+              await new Promise((resolve) => setTimeout(resolve, (2 ** attempt) * 1000));
```

Comment:
I would put this Promise in a helper method.

Reply:
Created a `delay` utility function. As a side effect, tests now run faster since it can be stubbed. Also, while refactoring this I found that iTwinPlatform uses a custom retry-after header, so I added handling for that as well.